notebook/services/eventlog/handlers.py

```diff
@@ -0,0 +1,42 @@
+import json
+
+from tornado import web
+
+from notebook.utils import url_path_join
+from notebook.base.handlers import APIHandler, json_errors
+from jupyter_telemetry.eventlog import EventLog
+
+class EventLoggingHandler(APIHandler):
+    """
+    A handler that receives and stores telemetry data from the client.
+    """
+    @json_errors
+    @web.authenticated
+    def post(self, *args, **kwargs):
+        try:
+            # Parse the data from the request body
+            raw_event = json.loads(self.request.body.strip().decode())
+        except Exception as e:
+            raise web.HTTPError(400, str(e))
+        
+        required_fields = {'schema', 'version', 'event'}
+        for rf in required_fields:
+            if rf not in raw_event:
+                raise web.HTTPError(400, '{} is a required field'.format(rf))
+
+        schema_name = raw_event['schema'] 
+        version = raw_event['version']
+        event = raw_event['event']
+        
+        # Profile, may need to move to a background thread if this is problematic
+        try: 
+            self.eventlog.record_event(schema_name, version, event)
+        except Exception as e:
+            raise web.HTTPError(400, e)
+            
+        self.set_status(204)
```

Comment:
nitpick: 204 is more common for PUTs, but do we have a reason for 200 vs 204?

Reply:
My original thinking was that we'd want `record_event` to be in a background thread, so we can return immediately after the request, without waiting for the sequence of events from `record_event` (which might include further HTTP requests to the ultimate backend!) to complete. This means we won't actually know if the request was a success - since we returned early. This is why I had this return 204 rather than 200.

I also considered making this a PUT but realized early on it isn't a good match since we're posting to `/api/eventlog` rather than `/api/eventlog/schema`